recipe/deploy/vendors.php

```diff
@@ -20,29 +20,48 @@
 /**
  * Set this variable to stable, snapshot, preview, 1 or 2 to select which Composer channel to use
  */
-set('composer_channel', null);
+set('composer_channel', 'stable');
 
 set('bin/composer', function () {
+    $composerVersionToInstall = get('composer_version');
+    $composerChannelToInstall = get('composer_channel');
+
+    $composerBin = null;
     if (commandExist('composer')) {
-        return '{{bin/php}} ' . locateBinaryPath('composer');
+        $composerBin = '{{bin/php}} ' . locateBinaryPath('composer');
     }
 
     if (test('[ -f {{deploy_path}}/.dep/composer.phar ]')) {
-        return '{{bin/php}} {{deploy_path}}/.dep/composer.phar';
+        $composerBin = '{{bin/php}} {{deploy_path}}/.dep/composer.phar';
     }
 
-    $composerVersionToInstall = get('composer_version', null);
-    $composerChannel = get('composer_channel', null);
-    $installCommand = "cd {{release_path}} && curl -sS https://getcomposer.org/installer | {{bin/php}}";
+    if ($composerBin) {
+        $currentComposerVersionRaw = run($composerBin . ' --version');
```

Comment:
This code is complicated. Please, add comments. 
Also, do we really need to support channel install? I'd like to keep code simple, what is the point in channel installs? I'm for vomiting this part., let's by default install from stable, or specified version if specified.

Reply:
What do you mean by " install from stable" - from channel named "stable"?

Reply:
Plz look here: https://getcomposer.org/versions
This are the only available information we have to figure out if we need to download composer or not.
So if I understand well if we install composer the standard way without any channel option then we install from "stable" channel.

Reply:
Correct.

Reply:
So if you want to support only "stable" channel then the only simplifing I can do is to remove line 47-49 - this line is support for snapshot channel.

Reply:
I added some comments. Please give me info if this can be final code so I can write some tests.